server/src/main/java/org/apache/druid/rpc/indexing/SpecificTaskServiceLocator.java

```diff
@@ -199,4 +208,20 @@ public void close()
       }
     }
   }
+
+  private class TaskLocationFetcher
```

Comment:
Ideally, this wrapper should have the whole logic of calling the first API, handling its future and then calling the second API.

Reply:
Edit: Resolving this as doing this would be much more involved for this PR.